src/block/inventory/SmithingTableInventory.php

```diff
@@ -25,13 +25,39 @@
 
 use pocketmine\inventory\SimpleInventory;
 use pocketmine\inventory\TemporaryInventory;
+use pocketmine\item\Item;
+use pocketmine\Server;
 use pocketmine\world\Position;
 
 final class SmithingTableInventory extends SimpleInventory implements BlockInventory, TemporaryInventory{
 	use BlockInventoryTrait;
 
+	public const SLOT_INPUT = 0;
+
+	public const SLOT_ADDITION = 1;
+
+	public const SLOT_TEMPLATE = 2;
+
 	public function __construct(Position $holder){
 		$this->holder = $holder;
-		parent::__construct(3);
+		parent::__construct(4);
+	}
+
+	public function getInput() : Item{
+		return $this->getItem(self::SLOT_INPUT);
+	}
+
+	public function getAddition() : Item{
+		return $this->getItem(self::SLOT_ADDITION);
+	}
+
+	public function getTemplate() : Item{
+		return $this->getItem(self::SLOT_TEMPLATE);
+	}
+
+	public function getOutput() : ?Item{
+		$craftingManager = Server::getInstance()->getCraftingManager();
+		$recipe = $craftingManager->matchSmithingRecipe($this->getInput(), $this->getAddition(), $this->getTemplate());
+		return $recipe?->constructOutput($this->getInput(), $this->getAddition(), $this->getTemplate());
 	}
```

Comment:
This doesn't belong here. It's not relevant to the function of the class.

Reply:
My comments from Discord: 
> just delete jt
if the handling was implemented correctly it would be unused anyway
look at how the CraftingTransaction works in ItemStackRequestExecutor
also, keep conversations in one place, it becomes a confusing mess of back-and-forth otherwise